ratis-server/src/main/java/org/apache/ratis/server/leader/LogAppenderDefault.java

```diff
@@ -73,9 +74,10 @@ private AppendEntriesReplyProto sendAppendEntriesWithRetries()
         }
 
         resetHeartbeatTrigger();
+        final Timestamp sendTime = Timestamp.currentTime();
         getFollower().updateLastRpcSendTime(request.getEntriesCount() == 0);
         final AppendEntriesReplyProto r = getServerRpc().appendEntries(request);
-        getFollower().updateLastRpcResponseTime();
+        getFollower().updateLastAppendEntriesResponseTime(sendTime);
```

Comment:
It seems that you should only update `lastRpcResponseTime` here when sending the RPC rather than updating your newly added field `lastRespondedAppendEntriesSendTime` because their meanings are different.

Reply:
Done.